### PR TITLE
Add a new AP_HAL_Linux Led_Sysfs class and use it for Disco

### PR DESCRIPTION
**Pull request by mothacehe**
In new Disco releases, the linux kernel led sysfs api will replace the pwm sysfs
api to drive the tricolor led.

Keep pwm sysfs api for compatibility reasons.

### CODE REVIEW

File: libraries/AP_HAL_Linux/Led_Sysfs.cpp [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        close(_brightness_fd);
+}
+
+bool Led_Sysfs::init()
+{
+    int r;
+    char *br_path;
+    char *max_br_path;
+
+    r = asprintf(&br_path, "/sys/class/leds/%s/brightness", _led_name);
```
Comment (khancyr) [inline libraries/AP_HAL_Linux/Led_Sysfs.cpp]: make direct assigment of r here !

File: libraries/AP_HAL_Linux/Led_Sysfs.cpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        AP_HAL::panic("LinuxLed_Sysfs : Couldn't allocate max_brightness path\n");
+    }
+
+    if (Util::from(hal.util)->read_file(max_br_path, "%u", &_max_brightness) < 0) {
+        AP_HAL::panic("LinuxLed_Sysfs : Unable to read max_brightness in %s\n",
+                      max_br_path);
+    }
+
+    free(max_br_path);
+
```
Comment (khancyr) [inline libraries/AP_HAL_Linux/Led_Sysfs.cpp]: no need to free br_path ? (not expert on that)

File: libraries/AP_HAL_Linux/Led_Sysfs.cpp [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+bool Led_Sysfs::set_brightness(uint8_t brightness)
+{
+    int br;
+
+    if (_brightness_fd <= 0)
+        return false;
+
+    br = brightness * _max_brightness / UINT8_MAX;
```
Comment (khancyr) [inline libraries/AP_HAL_Linux/Led_Sysfs.cpp]: create br only here and assign directly

File: libraries/AP_HAL_Linux/Led_Sysfs.cpp [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+Led_Sysfs::Led_Sysfs(const char *led_name)
+    : _led_name(led_name)
+{
+}
+
+Led_Sysfs::~Led_Sysfs()
+{
+    if (_brightness_fd)
+        close(_brightness_fd);
```
Comment (khancyr) [inline libraries/AP_HAL_Linux/Led_Sysfs.cpp]: I personnaly prefer to always use parenthesis, it prevents future and indentation problem.

File: libraries/AP_HAL_Linux/Led_Sysfs.cpp (hunk 1 above)
Comment (mothacehe) [inline libraries/AP_HAL_Linux/Led_Sysfs.cpp]: ok.

File: libraries/AP_HAL_Linux/Led_Sysfs.cpp (hunk 2 above)
Comment (mothacehe) [inline libraries/AP_HAL_Linux/Led_Sysfs.cpp]: yes you're right !

File: libraries/AP_HAL_Linux/Led_Sysfs.cpp (hunk 3 above)
Comment (mothacehe) [inline libraries/AP_HAL_Linux/Led_Sysfs.cpp]: ok.

File: libraries/AP_HAL_Linux/Led_Sysfs.cpp (hunk 4 above)
Comment (mothacehe) [inline libraries/AP_HAL_Linux/Led_Sysfs.cpp]: done.

File: libraries/AP_HAL_Linux/Led_Sysfs.cpp [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+    free(max_br_path);
+    free(br_path);
+
+    return true;
+}
+
+bool Led_Sysfs::set_brightness(uint8_t brightness)
+{
+    if (_brightness_fd <= 0) {
```
Comment (lucasdemarchi) [inline libraries/AP_HAL_Linux/Led_Sysfs.cpp]: 0 is a valid fd file. The check needs to be `< 0` and the initialization needs to be set to -1.

File: libraries/AP_HAL_Linux/Led_Sysfs.h [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+#include <AP_HAL/AP_HAL.h>
+
+#include "AP_HAL_Linux.h"
+#include "Util.h"
+
+#pragma once
+
+namespace Linux {
+    class Led_Sysfs {
```
Comment (lucasdemarchi) [inline libraries/AP_HAL_Linux/Led_Sysfs.h]: remove namespace indentation.

File: libraries/AP_HAL_Linux/Led_Sysfs.h [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        bool init();
+        bool set_brightness(uint8_t brightness);
+
+        Led_Sysfs(const char* led_name);
+        ~Led_Sysfs();
+
+    private:
+        int _brightness_fd;
+        int _max_brightness;
+        const char *_led_name;
```
Comment (lucasdemarchi) [inline libraries/AP_HAL_Linux/Led_Sysfs.h]: even if not required, please provide the initialization for these members. _brightness_fd needs to be initialized to -1 as mentioned above.

File: libraries/AP_HAL_Linux/Led_Sysfs.cpp (hunk 5 above)
Comment (mothacehe) [inline libraries/AP_HAL_Linux/Led_Sysfs.cpp]: ok, now fixed !

File: libraries/AP_HAL_Linux/Led_Sysfs.h (hunk 6 above)
Comment (mothacehe) [inline libraries/AP_HAL_Linux/Led_Sysfs.h]: ok

File: libraries/AP_HAL_Linux/Led_Sysfs.h (hunk 7 above)
Comment (mothacehe) [inline libraries/AP_HAL_Linux/Led_Sysfs.h]: ok

**Comment by mothacehe** [comment]
Lucas ?

**Comment by lucasdemarchi** [comment]
@mothacehe Applied, thanks.

I added some fixes on top, please take a look on those.

**Pull request closed without merging**